platforms/software/testing-base/src/main/java/org/gradle/api/tasks/testing/TestEventReporter.java

```diff
@@ -60,7 +65,30 @@ public interface TestEventReporter extends AutoCloseable {
      * @param value the metadata value, which must be serializable by the Tooling API
      * @since 8.12
      */
-    void metadata(Instant logTime, String key, Object value);
+    default void metadata(Instant logTime, String key, Object value) {
```

Comment:
💭 Moving the validations to this interface seems like it's not worth the extra complexity of adding default implementations if there is only a single (direct) implementation of this interface?